src/operations.jl

```diff
@@ -5,25 +5,9 @@ hash(c::AbstractRGB, hx::UInt) = hash(blue(c), hash(green(c), hash(red(c), hx)))
 hash(c::TransparentRGB, hx::UInt) = hash(alpha(c), hash(blue(c), hash(green(c), hash(red(c), hx))))
 
 # rand
-typealias RandTypes{T,C} Union{AbstractRGB{T},
-                               TransparentRGB{C,T},
-                               AbstractGray{T},
-                               TransparentGray{C,T}}
-typealias RandTypesFloat{C,T<:AbstractFloat} RandTypes{T,C}
-
-rand{G<:AbstractGray}(::Type{G}) = G(rand())
-rand{G<:TransparentGray}(::Type{G}) = G(rand(), rand())
-rand{C<:AbstractRGB}(::Type{C}) = C(rand(), rand(), rand())
-rand{C<:TransparentRGB}(::Type{C}) = C(rand(), rand(), rand(), rand())
-function rand{C<:RandTypesFloat}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(eltype(C), (sizeof(C)÷sizeof(eltype(C)), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f8}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt8, (sizeof(C), sz...)), sz)
-end
-function rand{C<:RandTypes{N0f16}}(::Type{C}, sz::Dims)
-    reinterpret(C, rand(UInt16, (sizeof(C)>>1, sz...)), sz)
-end
+rand{T<:Colorant}(::Type{T}) = reinterpret(T, rand(eltype(T), div(sizeof(T), sizeof(eltype(T)))))[1]
+rand{T<:Colorant}(::Type{T}, sz::Dims) =
+      reinterpret(T, rand(eltype(T), (div(sizeof(T), sizeof(eltype(T))), sz...)), sz)
```

Comment:
`rand(RGB)` works for me locally, and it's in the tests, which pass.  how am i confused?

```
julia> rand(RGB{Float64},3)
3-element Array{ColorTypes.RGB{Float64},1}:
 RGB{Float64}(0.735732,0.836881,0.282634)
 RGB{Float64}(0.987138,0.619426,0.803145)
 RGB{Float64}(0.65319,0.329882,0.335785) 

julia> rand(RGB{Float64})
RGB{Float64}(0.5079139816107203,0.9483896058733416,0.40892847391611986)
```